app/controllers/catalog_controller.rb

```diff
@@ -2059,7 +2059,7 @@ def replace_right_cell(options = {})
             }
           end
         end
-        if Settings.product.old_dialog_user_ui || action_name != "svc_catalog_provision"
+        if Settings.product.old_dialog_user_ui && action == "dialog_provision"
```

Comment:
I think this solves the issue of double buttons but the `&& action == "dialog_provision"` is unecessary, since we're already in an `elsif` block that has determined that `action == "dialog_provision"`. This would then change this to just `if Settings.product.old_dialog_user_ui`, which I think is safe to do since there's no other action that should go through this path.

Reply:
@eclarizio you are right, made suggested changes